dist/rename.py

```diff
@@ -66,14 +72,127 @@ def get_episode_from_media(filename: str, seasons: dict[str, int]) -> Optional[E
             show=SHOW_NAME,
             season=season_number,
             number=episode_number,
+            extended=match.group(4) is not None,
+            filepath=filepath,
         )
 
 
 def debugger_is_active() -> bool:
     return hasattr(sys, "gettrace") and sys.gettrace() is not None
 
 
+def ensure_tag_value(root, tag, value):
+    if root.find(tag) is None:
+        el = ET.Element(tag)
+        el.text = value
+        root.append(el)
+        return True
+    if root.find(tag).text != value:
+        root.find(tag).text = value
+        return True
+    return False
+
+
+def fix_season_nfo(fpath: Path, sno: int, sname: str):
+    try:
+        tree = ET.parse(str(fpath.absolute()))
+    except Exception as e:
+        print(fpath, e)
+        return
+    root = tree.getroot()
+    to_remove = []
+    for child in root:
+        if child.tag in [
+            "originaltitle",
+            "id",
+            "ratings",
+            "userrating",
+            "playcount",
+            "runtime",
+            "mpaa",
+            "watched",
+            "trailer",
+            "dateadded",
+            "epbookmark",
+            "code",
+            "fileinfo",
+            "source",
+            "original_filename",
+            "user_note",
+            "displayseason",
+            "displayepisode",
+        ]:
+            to_remove.append(child)
+            continue
+        if child.text is None:
+            to_remove.append(child)
+            continue
+    for child in to_remove:
+        root.remove(child)
+    edited = len(to_remove) > 0
+    if root.tag == "season":
+        edited = ensure_tag_value(root, "title", f"{sno}. {sname}") or edited
+        edited = ensure_tag_value(root, "seasonnumber", str(sno)) or edited
+    if edited:
+        tree.write(str(fpath.absolute()), xml_declaration=True, encoding="UTF-8")
```

Comment:
Adding `ET.indent(tree)` will fix the formatting issues in the NFO files (leading whitespace for last closing tag and whatever is going on in the recent s36 eps)

Reply:
Thanks for both bits you noticed, I should have fixed them now